demo/csi/digitalocean/plugin.nomad

```diff
@@ -10,16 +10,29 @@ job "digitalocean" {
     task "plugin" {
       driver = "docker"
 
-      config {
-        image = "digitalocean/do-csi-plugin:v2.1.1"
-        args = [
-          "--endpoint=${CSI_ENDPOINT}",
-          "--token=${token}",
-          "--url=https://api.digitalocean.com/",
-        ]
 
+      config {
+        image = "digitalocean/do-csi-plugin:v4.12.0"
+        entrypoint = [ "ash", "/local/run.sh" ]
         privileged = true
       }
+      
+      template {
+        data        = <<EOH
+DO_TOKEN="{{ with nomadVar "secrets/digitalocean_csi_driver" }}{{ .token }}{{ end }}"
+EOH
+        destination = "secrets/do-token.env"
+        env         = true
+      }
```

Comment:
This is an improved workflow for the plugin, but it breaks the demo because:
* You haven't granted the job an ACL that accesses this variable
* You haven't created the variable

Reply:
Actually, it'd be easier if you used a variable that's in one of the paths defined in the [Task Access to Variables](https://developer.hashicorp.com/nomad/docs/concepts/variables#task-access-to-variables) docs, that way you could avoid creating the ACL token and just use workload identity.